### PR TITLE
gidim/ add  'get_or_create_dataset()' method

### PR DESCRIPTION
**Pull request by gidim**
## Details
The process of creating dataset is iterative. At the moment it's super painful because every time you run your code you have to check whether the dataset exists or not and then decide if to create it. The goal is to make this seamless to the user.

In addition to the the above we will need to add backend functionality that doesn't that check for duplicates before saving a new dataset item

## Issues

Resolves #

## Testing
@alexkuzmik please let me know how you want me to test this (e2e/unit with mocks)

### CODE REVIEW

File: sdks/python/src/opik/api_objects/opik_client.py [hunk 1]
```diff
@@ -353,6 +355,25 @@ def create_dataset(
...
+
+        Args:
+            name: The name of the dataset.
+            description: An optional description of the dataset.
+
+        Returns:
+            dataset.Dataset: The dataset object.
+        """
+        try:
+            return self.get_dataset(name)
```
Comment (andrescrz) [inline sdks/python/src/opik/api_objects/opik_client.py:383]: Happy path requires a test.

File: sdks/python/src/opik/api_objects/opik_client.py [hunk 2]
```diff
@@ -353,6 +355,25 @@ def create_dataset(
...
+        except ApiError as e:
+            if e.status_code == 404:
+                return self.create_dataset(name, description)
```
Comment (andrescrz) [inline sdks/python/src/opik/api_objects/opik_client.py:386]: 404 error requires a test.

**Review by andrescrz — COMMENTED** [review]
Once tests are in place, this is good to go.

However, I agree that it'd be better to handle this in the backend. 

@thiagohora could you please add a ticket in our backlog for this? The create endpoint should create the dataset or return it if already exists by name as it's an unique key within the workspace. Any other conflict, it should fail.

**Review by ferc — APPROVED** [review]
Let's add the tests in the next PR, we can sync with @alexkuzmik later the best way to add them :)

**Review by ferc — APPROVED** [review]

**Comment by thiagohora** [comment]
> ? The create endpoint should create the dataset or return it if already exists by name as it's an unique key within the workspace. Any other conflict, it should fail.

Created ticket OPIK-209

**Pull request merged**